crates/sp-domains-fraud-proof/src/host_functions.rs

```diff
@@ -118,6 +116,38 @@ where
         .map(|ext| ext.encode())
     }
 
+    fn derive_domain_set_code_extrinsic(
+        &self,
+        consensus_block_hash: H256,
+        domain_id: DomainId,
+    ) -> Option<Option<Vec<u8>>> {
+        let maybe_upgraded_runtime = maybe_runtime_upgrade::<_, _, DomainBlock>(
+            &self.consensus_client,
+            consensus_block_hash.into(),
+            domain_id,
+        )
+        .ok()
+        .flatten();
+
+        if let Some(upgraded_runtime) = maybe_upgraded_runtime {
+            let runtime_code = self.get_domain_runtime_code(consensus_block_hash, domain_id)?;
+            let domain_runtime_api_light =
+                RuntimeApiLight::new(self.executor.clone(), runtime_code.into());
+
+            SetCodeConstructor::<DomainBlock>::construct_set_code_extrinsic(
+                &domain_runtime_api_light,
```

Comment:
Why explicit call and not `domain_runtime_api_light.construct_set_code_extrinsic()`?

Reply:
I wanted to do that but
`SetCodeConstructor` trait [here](https://github.com/subspace/subspace/blob/main/domains/client/block-preprocessor/src/runtime_api.rs#L34) takes a generic Domain `Block` type and it was not possible to pass that type without explicit type casting. So chose this route.

Ofcourse, we can move the generic type to trait function instead of trait itself but on the outside I felt it did not make much difference. Do we prefer that approach ?

Reply:
It is fine as is, I was just curious and API is odd